core/src/main/java/hudson/slaves/OfflineCause.java

```diff
@@ -52,6 +53,8 @@
      * {@link OfflineCause} that renders a static text,
      * but without any further UI.
      */
+    static Date timestamp;
```

Comment:
This should be declared `private final` (and non-`static`), initialized in no-arg constructor of `OfflineCause` and exposed using getter yielding defensive copy (as `Date` is not immutable).